### ISSUE TITLE
App Insights can't track users

### ISSUE THREAD

**Issue by Mahmoud-Hesham99** [Azure-Samples/azure-search-openai-demo#1714]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> Used manual steps to set up authentication, login works fine but app insights show only 1 user using the app and it's not the case.

### Any log messages given by the failure
>

### Expected/desired behavior
> Application Insights show the actual number of users instead of just 1 user.

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)

### azd version?
> run `azd version` and copy paste here.

### Versions
> 1.9.3

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by pamelafox** [comment]
Can you point at documentation about how App Insights tracks users? I wasn't aware that there was a feature for that.

**Comment by Mahmoud-Hesham99** [comment]
Here: 
- https://learn.microsoft.com/en-us/azure/azure-monitor/app/usage-segmentation
- https://github.com/Huachao/azure-content/blob/master/articles/application-insights/app-insights-web-track-usage.md 

and also you can easily find it on the portal, left panel -> Usage - > Users 

<img width="957" alt="App Insights - Users" src="https://github.com/Azure-Samples/azure-search-openai-demo/assets/73784370/c466c3a6-bb47-42ce-ab3c-288f895070f0">

**Comment by pope-llamas0g** [comment]
wondering about the same.

**Comment by pope-llamas0g** [comment]
Hi @pamelafox -  What would be the best way to track usage?

**Comment by mo-albaba** [comment]
@pamelafox , Yes I have (AZURE_USE_AUTHENTICATION="true") b ut as I mentioned on #2130 , it will display 1 user engagement on application insights for me as well.

**Comment by pamelafox** [comment]
I'm still figuring this out, I think we'll need something like this code:

    with tracer.start_as_current_span(name="get-job-span") as get_job_span:

        # Update the operation parameters
        get_job_span.set_attribute(ContextTagKeys.AI_USER_AUTH_USER_ID, users_oid)

But I'm not sure yet how to integrate that with configure_azure_monitor. We'd presumably need to call this for each request, potentially in our auth decorator.

**Comment by pamelafox** [comment]
We might need to bring in @tonybaloney's change:
https://github.com/Azure-Samples/azure-search-openai-demo/pull/1489/files#diff-d253a56fcf84fc8803ae3cdbe68e14f9861e4f0024bd0cfb3863c246aaf1a6af

**Comment by pope-llamas0g** [comment]
Hi @pamelafox yes, I was using authentication.

**Comment by jeannotdamoiseaux** [comment]
I encountered this issue a few days ago as well. The number of users on a given day is one of the most important metrics to track, if not the most important.

**Comment by pamelafox** [comment]
Okay, I'm reaching out to App Insights colleagues for tips on tracking users.

**Comment by jeremydvoss** [comment]
Thanks for reaching out. User monitoring is handled by the [client-side Web.js SDK](https://learn.microsoft.com/en-us/azure/azure-monitor/app/javascript-sdk?tabs=javascriptwebsdkloaderscript). You can set this up on top of the server-side Python SDK you already have.

**Comment by pamelafox** [comment]
Thanks @jeremydvoss - it looks we need to decide how to pass the connection string from the server to the client. We can either send it down with /authSetup, since currently we only want to use it when we're using authentication anyway, or send it down with /config. 

I think we may want to pass it down with /config, set it up for everyone, and then call the function for tracking based off user ID when we're in an authenticated environment:
https://learn.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics#authenticated-users

**Comment by ParthSharma023** [comment]
@pamelafox has there been any development on this?

**Comment by cforce** [comment]
Why don't we just keep using consistently open telemetry which out of the supports RUM https://www.npmjs.com/package/@opentelemetry/instrumentation-user-interaction
https://opentelemetry.io/docs/languages/js/getting-started/browser/#:~:text=Add%20Instrumentations

* RUM via OpenTelemetry JS SDK in the React frontend (frontend/)
* Distributed tracing in the Python FastAPI backend (app/backend/)
* Full context propagation (traceparent) from browser to backend
* No custom OpenTelemetry Collector — use ACA's built-in agent
* Export everything to Application Insights via OTLP (HTTP/gRPC)

Frontend Instrumentation (frontend/)

```
npm install --save \
  @opentelemetry/api \
  @opentelemetry/sdk-trace-web \
  @opentelemetry/exporter-trace-otlp-http \
  @opentelemetry/instrumentation-document-load \
  @opentelemetry/instrumentation-user-interaction \
  @opentelemetry/instrumentation-fetch \
  @opentelemetry/instrumentation-xml-http-request
```

```
import { WebTracerProvider } from '@opentelemetry/sdk-trace-web';
import { BatchSpanProcessor } from '@opentelemetry/sdk-trace-base';
import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
import {
  DocumentLoadInstrumentation,
  UserInteractionInstrumentation,
  FetchInstrumentation,
  XMLHttpRequestInstrumentation
} from '@opentelemetry/instrumentation-*';

const provider = new WebTracerProvider();
provider.addSpanProcessor(
  new BatchSpanProcessor(
    new OTLPTraceExporter({
      url: 'http://localhost:4318/v1/traces' // ACA agent OTLP HTTP endpoint
    })
  )
);
provider.register();

new DocumentLoadInstrumentation().enable();
new UserInteractionInstrumentation().enable();
new FetchInstrumentation().enable();
new XMLHttpRequestInstrumentation().enable();
```

Backend Instrumentation (app/backend/)

```
pip install \
  opentelemetry-api \
  opentelemetry-sdk \
  opentelemetry-exporter-otlp \
  opentelemetry-instrumentation-asgi \
  opentelemetry-instrumentation-requests
```

```
from opentelemetry import trace
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
from opentelemetry.sdk.trace.export import BatchSpanProcessor
from opentelemetry.instrumentation.asgi import OpenTelemetryMiddleware
from opentelemetry.instrumentation.requests import RequestsInstrumentor
from opentelemetry.sdk.resources import Resource

trace.set_tracer_provider(TracerProvider(resource=Resource.create({
    "service.name": "search-openai-backend"
})))
trace.get_tracer_provider().add_span_processor(
    BatchSpanProcessor(OTLPSpanExporter(endpoint="localhost:4317", insecure=True))
)

from app.main import app  
app.asgi_app = OpenTelemetryMiddleware(app.asgi_app)
RequestsInstrumentor().instrument()
```

The Trace context from browser → backend is automatically preserved via the traceparent HTTP header.
The Telemetry shall shows up in Application Insights
As we already use OpenTelemetr — this will extend that setup to emit RUM spans to the ACA agent.

**Comment by ram-atea** [comment]
Has there been any progress or implementation on tracking users?
